### PR TITLE
MULE-13590: scripting module fixes

### PR DESCRIPTION
**Pull request by alepulver**

### CODE REVIEW

File: src/main/java/org/mule/plugin/scripting/component/ScriptRunner.java [hunk 1]
```diff
@@ -52,41 +52,38 @@
...
   private CompiledScript compiledScript;
 
   private ScriptEngine scriptEngine;
   private ScriptEngineManager scriptEngineManager;
 
-  public ScriptRunner(Script scriptConfig, MuleContext muleContext) {
-    this.scriptConfig = scriptConfig;
-    this.muleContext = muleContext;
+  public ScriptRunner(String engine, String code, ComponentLocation location) {
+    this.engineText = engine;
```
Comment (elrodro83) [inline src/main/java/org/mule/plugin/scripting/component/ScriptRunner.java]: why not just `engine` or `engineName`? Putting `text` here can cause confusion

File: src/main/java/org/mule/plugin/scripting/component/ScriptRunner.java [hunk 2]
```diff
@@ -52,41 +52,38 @@
...
+    this.codeText = code;
```
Comment (elrodro83) [inline src/main/java/org/mule/plugin/scripting/component/ScriptRunner.java]: `scriptBody`?

File: src/main/java/org/mule/plugin/scripting/component/ScriptRunner.java [hunk 3]
```diff
@@ -159,4 +153,18 @@ protected String listAvailableEngines() {
...
+    if (value instanceof CursorProvider) {
+      value = ((CursorProvider) value).openCursor();
+    }
+
+    return value;
+  }
+
+  private Map<String, Object> createResolvedMap(CoreEvent event) {
+    return event.getVariables().entrySet().stream().collect(
+                                                            Collectors.toMap(e -> e.getKey(),
```
Comment (elrodro83) [inline src/main/java/org/mule/plugin/scripting/component/ScriptRunner.java]: static import

File: src/test/resources/groovy-component-config-flow.xml [hunk 4]
```diff
@@ -73,10 +73,11 @@
     <flow name="inlineScriptMutatePayload">
         <script:execute engine="groovy">
             <script:code>
-                payload = 'bar-mutated'               
+                payload = 'bar-mutated'
```
Comment (elrodro83) [inline src/test/resources/groovy-component-config-flow.xml]: is setting the payload as variable no longer valid?

Comment (alepulver) [reply src/test/resources/groovy-component-config-flow.xml]: Right. Assignment is no longer possible to bound variables, only method calls.

**Pull request merged**